storage/badger/receipts.go

```diff
@@ -140,16 +140,29 @@ func (r *ExecutionReceipts) Index(blockID, receiptID flow.Identifier) error {
 		}
 
 		return nil
-	})
+	}
+	r.byBlockId = func(tx *badger.Txn, blockID flow.Identifier) ([]*flow.ExecutionReceipt, error) {
+		var receiptID flow.Identifier
+		err := operation.LookupExecutionReceipt(blockID, &receiptID)(tx)
+		if err != nil {
+			return nil, fmt.Errorf("could not lookup receipt ID: %w", err)
+		}
```

Comment:
Yeah, as I said, really not sure about it. See my [comment](https://github.com/onflow/flow-go/pull/486#discussion_r587822748). I don't have any preference so can do either way.

Reply:
I would suggest to return empty slice instead of `storage.NotFoundError`, because:
1) [access node is expecting to return empty slice here](https://github.com/onflow/flow-go/pull/486/files#diff-ac13d40a77a065ec65a685ed828d76f849a988759b47ce8c2b6f1a8750b7aed6R215-R228), if we return a not found error, it might produce a confusing error message.
1) `ByBlockID` is used by both ExecutionReceipts and MyExecutionReceipts, it's better to keep the behavior consistent by both returning empty slice when receipts are not found by the block ID.